packages/syft/tests/syft/metadata/metadata_service_test.py

```diff
@@ -0,0 +1,203 @@
+
+    response = metadata_service.get(context=authed_context)
+    assert isinstance(response, SyftError)
+    assert response.message == mock_error_message
+
+
+def test_metadataservice_set_success(
+    metadata_service: MetadataService,
+    metadata: NodeMetadata,
+    authed_context: AuthedServiceContext,
+) -> None:
+    response = metadata_service.set(authed_context, metadata)
+
+    assert response.is_ok() is True
+    assert isinstance(response.ok(), NodeMetadata)
+    assert response.ok() == metadata
+
+
+def test_metadataservice_set_fail(
+    monkeypatch: MonkeyPatch,
+    metadata_service: MetadataService,
+    metadata: NodeMetadata,
+    authed_context: AuthedServiceContext,
+) -> None:
+    mock_error_message = "database failure"
+
+    def mock_stash_set_error(a) -> Err:
+        return Err(mock_error_message)
+
+    monkeypatch.setattr(metadata_service.stash, "set", mock_stash_set_error)
+
+    response = metadata_service.set(authed_context, metadata)
+
+    assert isinstance(response, SyftError)
+    assert response.message == mock_error_message
+
+
+def add_mock_metadata(
+    metadata_stash: MetadataStash, metadata: NodeMetadata
+) -> NodeMetadata:
+    # create a mock metadata in the stash so that we can update it
+    result = metadata_stash.partition.set(metadata)
+    assert result.is_ok()
+
+    created_metadata = result.ok()
+    assert created_metadata is not None
+
+    return created_metadata
+
+
+def test_metadataservice_update_success(
+    monkeypatch: MonkeyPatch,
+    metadata_stash: MetadataStash,
+    metadata_service: MetadataService,
+    metadata: NodeMetadata,
+    update_metadata: NodeMetadataUpdate,
+    authed_context: AuthedServiceContext,
+) -> None:
+    # add a mock metadata to the stash
+    mock_metadata = add_mock_metadata(metadata_stash, metadata)
+
+    # get a new metadata according to update_metadata
+    new_metadata = deepcopy(metadata)
+    update_kwargs = update_metadata.to_dict(exclude_none=True).items()
+    for field_name, value in update_kwargs:
+        setattr(new_metadata, field_name, value)
+
+    assert new_metadata != metadata
+    assert new_metadata != mock_metadata
+    assert mock_metadata == metadata
+
+    mock_stash_get_all_output = [mock_metadata, mock_metadata]
+
+    def mock_stash_get_all() -> Ok:
+        return Ok(mock_stash_get_all_output)
+
+    monkeypatch.setattr(metadata_service.stash, "get_all", mock_stash_get_all)
+
+    # update the metadata in the metadata stash using metadata_service
+    response = metadata_service.update(authed_context, update_metadata)
+    updated_metadata = response.ok()[0]
+    not_updated_metadata = response.ok()[1]
+
+    assert response.is_ok() is True
+    assert len(response.ok()) == len(mock_stash_get_all_output)
+    assert updated_metadata == new_metadata  # the first metadata is updated
+    assert not_updated_metadata == metadata  # the second metadata is not updated
+
+    assert True
```

Comment:
we can remove this.

Reply:
I think this 2 lines 
`
assert updated_metadata == new_metadata  # the first metadata is updated
assert not_updated_metadata == metadata  # the second metadata is not updated
`
are necessary, no?

Reply:
I was just talking about the single `assert True`, not the rest of them.